server/build_event_protocol/build_status_reporter/build_status_reporter.go

```diff
@@ -71,6 +74,54 @@ func (r *BuildStatusReporter) initGHClient(ctx context.Context) interfaces.GitHu
 	return r.env.GetGitHubStatusService().GetStatusClient(accessToken)
 }
 
+func (r *BuildStatusReporter) isStatusReportingEnabled(ctx context.Context, repoURL string) bool {
+	r.once.Do(func() {
+		if dbh := r.env.GetDBHandle(); dbh != nil {
+			userInfo, err := r.env.GetAuthenticator().AuthenticatedUser(ctx)
+			if err != nil {
+				log.CtxWarningf(ctx, "Failed to report GitHub status, no authenticated user: %s", err)
+				return
+			}
+
+			parsedRepo, err := gitutil.ParseGitHubRepoURL(repoURL)
+			if err != nil {
+				log.CtxWarningf(ctx, "Failed to report GitHub status, invalid repo url %s: %s", repoURL, err)
+				return
+			}
+
+			installation := &tables.GitHubAppInstallation{}
+			err = dbh.NewQuery(ctx, "build_status_reporter_get_app_installation").Raw(
+				`SELECT * from "GitHubAppInstallations" WHERE group_id = ? AND owner = ?`, userInfo.GetGroupID(), parsedRepo.Owner).Take(installation)
+			if err == nil {
+				r.shouldReportCommitStatuses = installation.ReportCommitStatusesForCIBuilds
+				return
+			}
+
+			// If the user hasn't installed our GH app, check legacy methods for
+			// enabling status reporting. Always report statuses for users that
+			// onboarded through a legacy method, because status reporting was
+			// automatically enabled for them.
+			legacyWorkflow := &tables.Workflow{}
+			err = dbh.NewQuery(ctx, "build_status_reporter_get_workflow").Raw(
+				`SELECT * from "Workflows" WHERE repo_url = ?`, repoURL).Take(legacyWorkflow)
+			if err == nil {
+				r.shouldReportCommitStatuses = true
+				return
+			}
+
+			group := &tables.Group{}
+			err = dbh.NewQuery(ctx, "build_status_reporter_get_group").Raw(
+				`SELECT * from "Groups" WHERE group_id = ? AND github_token <> "" AND github_token IS NOT NULL`, userInfo.GetGroupID()).Take(group)
+			if err == nil {
+				r.shouldReportCommitStatuses = true
+				return
+			}
```

Comment:
can we distinguish between `db.IsRecordNotFound` and other DB errors for these 3 cases and maybe log the errors? or, could we select `COUNT` since we're not using the returned fields anyway (any errors returned from a COUNT query would be legit errors rather than NotFound, so we could unconditionally log the error)